### DISCUSSION TITLE
HandleBars get first item from array

### DISCUSSION THREAD

**Discussion by deiv061** [rfennell/AzurePipelines#1245] [Q&A]
Hello,

Everything works as well with the release definition and wiki page updater. I have thoughts in my mind I will give you more in my example.

For my wiki page, I have shown the Last PR, Commit Message, and Description
But in my description, I have a title ID from the Jira issue.
**, For example, #4777  Description Message** 
How can I get from this, the description is only the #4777 and visualized into a new cell in Wiki.
Let the description column be the whole message, but in the new cell, I want to build the URL for Jira Issues and put the object from the description message. 
| Description Message | Jira Link |
| -- | -- |
|#4777  Description Message | [4777](https://my.jira.link/4777)

Here is my template:

<table>
  <tr>
    <th>PullRequst ID</th>
    <th>Commit Message</th>
    <th>Description</th>
  </tr>
  {{#forEach pullRequests}}
  <tr>
<td>

 [{{this.pullRequestId}}]({{replace (replace this.url "_apis/git/repositories" "_git") "pullRequests" "pullRequest"}})
       
</td>
<td>{{this.title}}</td>

<td>{{this.description}}

</td>
  </tr>
  {{/forEach}}
</table>

**Comment by rfennell** [comment]
You would need to write a custom handlebars extension to take the description string and split it up to build the JIRA URL

See the [wiki](https://github.com/rfennell/AzurePipelines/wiki/GenerateReleaseNotes---Node-based-Cross-Platform-Task#handlebar-templates) for details, and the aresome [sample here](https://github.com/rfennell/AzurePipelines/tree/main/SampleTemplates/XplatGenerateReleaseNotes%20(Node%20based)/Version%203)

**Reply by deiv061** [reply]
Thank you, seems to work now but how can I foreach function's result?
<
const handlebars = require("handlebars");
module.exports = {
      filter_title_by_items(str) {
      const regex = /(#?(MS-?(?<issue2>[\d]{3,})))/gm;
      var jiraurl = "https://jira.domain.com/browse/MS-"
      var urls = [];
      var m;
      while ((m = regex.exec(str)) !== null) {
      if (m.index === regex.lastIndex) {
        regex.lastIndex++;
      }
       urls.push(jiraurl + m[m.length -1])
    }
       return urls;
  },
}
>
< 
<td>
  {{#forEach (filter_title_by_items this.title)}}
   {{this}}
  {{/forEach}}
</td>
>
Error: Error Processing handlebars [TypeError: Cannot create property 'index' on string ]
tried with {{this}}, {{this.urls}}
Because the title can have two Jira tasks numbers in associated PR

**Reply by rfennell** [reply]
You can write an extension that manages the for loop, have a look at [this sample](https://github.com/rfennell/AzurePipelines/blob/main/SampleTemplates/XplatGenerateReleaseNotes%20(Node%20based)/Version%203/each_with_sort_by_field.js) for the basic format

**Reply by rfennell** [reply]
I am no Handlebars expert, so best to google for resources on writing Handlebars extensions

**Reply by deiv061** [reply]
Everything works, as well if someone needs it. I will post it here a function with regex and generate a link to Jira issues
```
const handlebars = require("handlebars");
module.exports = {
      filter_title_by_items(title,desc) {
      let result = title.concat(desc);
      const regex = /(#?(MS-?(?<issue2>[\d]{3,})))/gm;
      var jiraurl = "https://jira-domain,com/browse/MS-"
      var urls = [];
      var regexresult;
      while ((regexresult = regex.exec(result)) !== null) {
      if (regexresult.index === regex.lastIndex) {
        regex.lastIndex++;
      }
        urls.push(`[MS-${regexresult[regexresult.length - 1]}](${jiraurl + regexresult[regexresult.length - 1]})`)
       }
       urls = urls.filter((x, i, a) => a.indexOf(x) == i)
       return urls;
  }
}

**Reply by rfennell** [reply]
Good news, glad it works for you

Feel free to submit a PR with the sample into the [/SampleTemplates/XplatGenerateReleaseNotes%20(Node%20based)/Version%203](https://github.com/rfennell/AzurePipelines/tree/main/SampleTemplates/XplatGenerateReleaseNotes%20(Node%20based)/Version%203) samples folder.

If you do can you follow the same format as the other `*.js` samples i.e a header containing usage, and update the `readme.md` in he folder as it is the index
